src/Microsoft.OData.Edm/EdmUtil.cs

```diff
@@ -492,6 +500,66 @@ internal static string FullyQualifiedName(IEdmVocabularyAnnotatable element)
             return null;
         }
 
+        private static string FullyQualifiedName(IEdmAnnotationsTarget annotationsTarget)
+        {
+            string qualifiedName = string.Empty;
+            List<IEdmElement> segments = annotationsTarget.TargetSegments.ToList();
+            IEdmEntityContainer container = segments[0] as IEdmEntityContainer;
+            IEdmEntityContainerElement containerElement = segments[1] as IEdmEntityContainerElement;
+
+            if (container == null || containerElement == null)
+            {
+                return null;
+            }
+
+            qualifiedName = container.FullName() + "/" + containerElement.Name;
+
+            if (segments.Count == 3)
+            {
+                // The only supported targets
+                // MySchema.MyEntityContainer/MyEntitySet/MyProperty
+                // MySchema.MyEntityContainer/MyEntitySet/MyNavigationProperty
+
+                IEdmProperty property = segments[2] as IEdmProperty;
+
+                qualifiedName = qualifiedName + "/" + property.Name;
+            }
+            else if (segments.Count == 4)

```

Comment:
>segments.Count == 4) [](http://example.com/codeflow?start=21&length=20)

we can't say there's on 3, 4 segments in the path. 
It could be any numbers in the segments.
For example:  "it could be "EntityContainer/set/property1/proeprty2/typecast1/property11/typecast2/propety...."

Reply:
Fixed